### PR TITLE
detect jwt policy and support first-party jwt token for VM

### PR DESCRIPTION
**Pull request by irisdingbj**
fix https://github.com/istio/istio/issues/30573

### CODE REVIEW

File: istioctl/cmd/workload.go [hunk 1]
```diff
@@ -333,6 +335,29 @@ func createCertsTokens(kubeClient kube.ExtendedClient, wg *clientv1alpha3.Worklo
 	}
 
 	serviceAccount := wg.Spec.Template.ServiceAccount
+	tokenPath := filepath.Join(dir, "istio-token")
+	jwtPolicy, err := util.DetectSupportedJWTPolicy(kubeClient.RESTConfig())
+	if err != nil {
+		fmt.Printf("Failed to determine JWT policy support: %v", err)
```
Comment (esnible) [inline istioctl/cmd/workload.go:341]: This is not your fault, but please don't use `fmt.Printf()` in istioctl.  Instead, pass `cmd.OutOrStdout()` as a `io.Writer` and use fmt.Fprintf(writer, ...)`.  Use _proxyconfig.go_ as an example.  The reason is that it makes Go tests easier to write.

File: istioctl/cmd/workload.go [hunk 2]
```diff
@@ -333,6 +335,29 @@ func createCertsTokens(kubeClient kube.ExtendedClient, wg *clientv1alpha3.Worklo
...
+	}
+	if jwtPolicy == util.FirstPartyJWT {
+		fmt.Println("warning: Detected that your cluster does not support third party JWT authentication. " +
```
Comment (esnible) [inline istioctl/cmd/workload.go:344]:
```suggestion
		fmt.Println("Warning: cluster does not support third party JWT authentication. " +
```

File: istioctl/cmd/workload.go [hunk 3]
```diff
@@ -333,6 +335,29 @@ func createCertsTokens(kubeClient kube.ExtendedClient, wg *clientv1alpha3.Worklo
...
+			return err
+		}
+		secret, err := kubeClient.CoreV1().Secrets(wg.Namespace).Get(context.TODO(), sa.Secrets[0].Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(tokenPath, secret.Data["token"], filePerms); err != nil {
+			return err
+		}
+		fmt.Printf("warning: a security token for namespace %s and service account %s has been generated and stored at %s\n", wg.Namespace, serviceAccount, tokenPath)
```
Comment (esnible) [inline istioctl/cmd/workload.go:358]:
```suggestion
		fmt.Printf("Warning: a security token for namespace %q and service account %q has been generated and stored at %s\n", wg.Namespace, serviceAccount, tokenPath)
```

I realize you were following the existing error message.  Either change both or ignore my suggestion.

**Review by esnible — COMMENTED** [review]
Overall looks good.  I suggest minor wording changes, and a minor refactor.

Feel free to ignore my suggestion to pass `cmd.OutOrStdout()` if it is too much work.  I looked at the existing tests in _workload_test.go_ and they do not cover the fmt.Printf() messages.

File: istioctl/cmd/workload.go (hunk 1 above)
Comment (irisdingbj) [reply istioctl/cmd/workload.go]: Sure, I have modified to use cmd.OutOrStdout now

File: istioctl/cmd/workload.go (hunk 2 above)
Comment (irisdingbj) [reply istioctl/cmd/workload.go]: done

File: istioctl/cmd/workload.go (hunk 3 above)
Comment (irisdingbj) [reply istioctl/cmd/workload.go]: changed both

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**